### ISSUE TITLE
Mesh not removed from scene when a node in its parent chain is dismount()ed

### ISSUE THREAD

**Issue by aprender** [Famous/engine#439]
<H1>Description</H1>

Calling dismount() on a Node() that is a parent of a Mesh() does not remove the mesh from a scene. Instead, the Mesh remains in the scene and any components along that same chain are no longer updated, thus any animated transformations applied to the mesh seem to "freeze".

I have encountered some other more obscure examples of generalised weirdness when dismount()ing meshes, but hopefully they are all part of this same bug. Will test a few things again once a fix comes through.

<H1>Steps to Reproduce</H1>
<OL>
<LI>Add a node X to a scene using `var x = scene.addChild()`</LI>
<LI>Add a `Mesh` to node X</LI>
<LI>Add a component to that mesh to animate it</LI>
<LI>Dismount node X `x.dismount()`</LI>
<LI>Observe that mesh remains in scene but animation of mesh stops</LI>
</OL>

<H1>Isolation</H1>

Problem exists in:
- 0.6.2
- 0.7.0
- develop as at 2015-07-21 (b2017d9c3577b2890d8cbfb807f0b3bc3ed4c4fd)
with @alexanderGugel's pull request entitled 'fix: Correctly mount and dismount' (700d2594d782b69af08c9417ec61469c1bd7afd5) applied.

<H1>Code Example</H1>

In this complete example, a wire-frame cube should spin for 3 seconds then disappear. Instead it freezes in place after 3 seconds.

See `oCubeNode.dismount();` below:

```HTML
<html>
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Famous0.6.2</title>
    <link rel="icon" href="favicon.ico?v=1" type="image/x-icon">
    <meta name="description" content="Seed Famous@0.6.2">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="Cache-Control" content="no-cache, no-store, must-revalidate" />
    <meta http-equiv="Pragma" content="no-cache" />
    <meta http-equiv="Expires" content="0" />
    <script src="http://code.famo.us/famous/0.6.2/famous.min.js"></script>

    <style>
    html, body {
        width: 100%;
        height: 100%;
        margin: 0px;
        padding: 0px;

    }
    body {
        position: absolute;
        -webkit-transform-style: preserve-3d;
        transform-style: preserve-3d;
        -webkit-font-smoothing: antialiased;
        -webkit-tap-highlight-color: transparent;
        background-color: black;
        -webkit-perspective: 0;
        perspective: none;
        overflow: hidden;
    }
    </style>
</head>

<body>
<script>
'use strict';

console.log(famous);
//var famous = require('famous');
var DOMElement = famous.domRenderables.DOMElement;
var FamousEngine = famous.core.FamousEngine;
var Mesh = famous.webglRenderables.Mesh;
var Color = famous.utilities.Color;
var Geometry = famous.webglGeometries.Geometry;

var scene = FamousEngine.createScene();
var centeredOriginNode = scene.addChild();

FamousEngine.init();

scene
    .setSizeMode('relative', 'relative', 'absolute')
    .setProportionalSize(0.7, 0.7, 0.7)
    .setAbsoluteSize(1000, 1000, 500)
    ;

centeredOriginNode =
        createNode(scene)
            //.setAbsoluteSize(110,110,110)
                .setAlign(0.5,0.5,0.5)
                .setMountPoint(0.5,0.5,0.5)
;

var oCubeNode = addDebugCube(centeredOriginNode);

var spinner = scene.addComponent({
    onUpdate: function(time) {
        centeredOriginNode.setRotation(time/250, time/1000, time/500);
        scene.requestUpdateOnNextTick(spinner);
    }
});

scene.requestUpdate(spinner);

setTimeout(function() {
    //centeredOriginNode.hide();        // this works (mesh disappears)
    //centeredOriginNode.dismount();    // DOES NOT WORK (MESH FREEZES IN PLACE)
    oCubeNode.dismount();             // DOES NOT WORK (MESH FREEZES IN PLACE)
}, 1000 * 3);



// utils

function addDebugCube(parent) {

    var oCube = parent.addChild().setOpacity(0.3);

    // cube
    var oGeometry =
            new Geometry({
                type: 'LINES',
                buffers: [
                    { name: 'a_pos', data: [
                        -1,-1,-1, 1,-1,-1, 1,1,-1, -1,1,-1,
                        -1,-1,1, 1,-1,1, 1,1,1, -1,1,1
                    ], size: 3 },
                    { name: 'indices', data: [
                        0,1, 1,2, 2,3, 3,0, // front face
                        4,5, 5,6, 6,7, 7,4, // back face
                        0,4, 4,7, 7,3, 3,0, // left face
                        1,5, 5,6, 6,2, 2,1 // right face
                    ], size: 2 }
                ]});

    var oMesh =
            new Mesh(oCube)
                    .setGeometry(oGeometry)
                    .setBaseColor(new Color("#FFFFFF"))
            ;

    return oCube;
};


function createNode(parent) {
    var node = parent.addChild();

    var aParentSizeMode = parent.getSizeMode();
    var aParentProportionalSize = parent.getProportionalSize();
    var aParentAbsoluteSize = parent.getAbsoluteSize();
    var aParentDifferentialSize = parent.getDifferentialSize();

    node
            .setSizeMode(aParentSizeMode[0], aParentSizeMode[1], aParentSizeMode[2])
            .setProportionalSize(aParentProportionalSize[0], aParentProportionalSize[1], aParentProportionalSize[2])
            .setAbsoluteSize(aParentAbsoluteSize[0], aParentAbsoluteSize[1], aParentAbsoluteSize[2])
            .setDifferentialSize(aParentDifferentialSize[0], aParentDifferentialSize[1], aParentDifferentialSize[2])
            .setOrigin(0.5, 0.5, 0.5)
    ;

    return node;
}
</script>
</body>
</html>
```

**Comment by alexanderGugel** [comment]
Thanks for reporting. We are aware of this issue and I'm working on a solution as we speak.

The issue is that the `WITH` command is currently not being sent in a lot of commands. This is a separate issue. The referenced PR only ensures that the state of the Mesh itself is updated correctly.

#436

**Comment by aprender** [comment]
Fab, yell out when you have something to test.

**Comment by alexanderGugel** [comment]
@aprender I made a quick fix. 6fff5ce should work for now. I'm working on a bigger refactor though.

**Comment by aprender** [comment]
Hi Alex,

Thanks for doing it so quick - I tested that fix and its a little too successful at deleting things! It very neatly resolves the problem described in the initial example as well as deleting everything from the scene.

So allow me to provide a revised piece of code ;)

<H1>Isolation</1>
Tested against your previous fix (700d2594d782b69af08c9417ec61469c1bd7afd5) plus your new one (6fff5ce66b1798cea77a741f4a85e96b22cb8ea2) merged into the latest develop HEAD.

<H1>Example Code</H1>

```JavaScript
<html>
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Famous0.6.2</title>
    <link rel="icon" href="favicon.ico?v=1" type="image/x-icon">
    <meta name="description" content="Seed Famous@0.6.2">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="Cache-Control" content="no-cache, no-store, must-revalidate" />
    <meta http-equiv="Pragma" content="no-cache" />
    <meta http-equiv="Expires" content="0" />
    <xxxscript src="http://code.famo.us/famous/0.6.2/famous.min.js"></xxxscript>
    <script src="lib/famous-20150721-b2017d9c3577b2890d8cbfb807f0b3bc3ed4c4fd-HACKED.js"></script>

    <style>
    html, body {
        width: 100%;
        height: 100%;
        margin: 0px;
        padding: 0px;

    }
    body {
        position: absolute;
        -webkit-transform-style: preserve-3d;
        transform-style: preserve-3d;
        -webkit-font-smoothing: antialiased;
        -webkit-tap-highlight-color: transparent;
        background-color: black;
        -webkit-perspective: 0;
        perspective: none;
        overflow: hidden;
    }
    </style>
</head>

<body>
<script>
'use strict';

console.log(famous);
//var famous = require('famous');
var DOMElement = famous.domRenderables.DOMElement;
var FamousEngine = famous.core.FamousEngine;
var Mesh = famous.webglRenderables.Mesh;
var Color = famous.utilities.Color;
var Geometry = famous.webglGeometries.Geometry;

var scene = FamousEngine.createScene();
var centeredOriginNode = scene.addChild();

FamousEngine.init();

scene
    .setSizeMode('relative', 'relative', 'absolute')
    .setProportionalSize(0.7, 0.7, 0.7)
    .setAbsoluteSize(1000, 1000, 500)
    ;


centeredOriginNode =
    createNode(scene)
    .setAlign(0.5,0.5,0.5)
    .setMountPoint(0.5,0.5,0.5)
    ;

spin(centeredOriginNode, 2500,10000,5000);

// add big cube
var oBigCubeNode = addDebugCube(centeredOriginNode);

// add some little cubes to it

var aCubeNodes = [];

for ( var i = 0; i < 100; i++ ) {
    var fCubeSize = Math.random() * 100;
    var oNewCube =
        addDebugCube(oBigCubeNode)
            .setSizeMode('absolute', 'absolute', 'absolute')
            .setProportionalSize(0.7, 0.7, 0.7)
            .setAbsoluteSize(fCubeSize, fCubeSize, fCubeSize)
            .setAlign(Math.random(), Math.random(), Math.random())
            .setOrigin(0.5, 0.5, 0.5)
            .setMountPoint(0.5, 0.5, 0.5)
            ;
    spin(oNewCube,(Math.random()-0.5) * 1000 + 100, (Math.random()-0.5) * 1000 + 100, (Math.random()-0.5) * 1000 + 100);
    aCubeNodes.push(oNewCube);
}

setTimeout(function() {
    //centeredOriginNode.hide();        // this works (mesh disappears)
    //centeredOriginNode.dismount();    // this works (mesh disappears)
    oBigCubeNode.dismount();            // this works (mesh disappears)
    //removeCube();                     // THIS DOESN'T WORK (EVERYTHING DISAPPEARS AT ONCE)
    //aCubeNodes[0].dismount();         // THIS DOESN'T WORK (EVERYTHING DISAPPEARS AT ONCE)
}, 1000 * 3);


// utils

function removeCube() {
    if (aCubeNodes.length <= 0)
        return;
    aCubeNodes.pop().dismount();
    setTimeout(removeCube, 100);
}

function spin(node, tx,ty,tz) {
    var spinner = node.addComponent({
        onUpdate: function(time) {
            node.setRotation(!tx ? 0 : time/tx, !ty ? 0 : time/ty, !tz ? 0 : time/tz);
            node.requestUpdateOnNextTick(spinner);
        }
    });
    node.requestUpdate(spinner);
}

function addDebugCube(parent) {

    var oCube = parent.addChild().setOpacity(0.3);

    // cube
    var oGeometry =
            new Geometry({
                type: 'LINES',
                buffers: [
                    { name: 'a_pos', data: [
                        -1,-1,-1, 1,-1,-1, 1,1,-1, -1,1,-1,
                        -1,-1,1, 1,-1,1, 1,1,1, -1,1,1
                    ], size: 3 },
                    { name: 'indices', data: [
                        0,1, 1,2, 2,3, 3,0, // front face
                        4,5, 5,6, 6,7, 7,4, // back face
                        0,4, 4,7, 7,3, 3,0, // left face
                        1,5, 5,6, 6,2, 2,1 // right face
                    ], size: 2 }
                ]});

    var oMesh =
            new Mesh(oCube)
                    .setGeometry(oGeometry)
                    .setBaseColor(new Color("#FFFFFF"))
            ;

    return oCube;
};


function createNode(parent) {
    var node = parent.addChild();

    var aParentSizeMode = parent.getSizeMode();
    var aParentProportionalSize = parent.getProportionalSize();
    var aParentAbsoluteSize = parent.getAbsoluteSize();
    var aParentDifferentialSize = parent.getDifferentialSize();

    node
            .setSizeMode(aParentSizeMode[0], aParentSizeMode[1], aParentSizeMode[2])
            .setProportionalSize(aParentProportionalSize[0], aParentProportionalSize[1], aParentProportionalSize[2])
            .setAbsoluteSize(aParentAbsoluteSize[0], aParentAbsoluteSize[1], aParentAbsoluteSize[2])
            .setDifferentialSize(aParentDifferentialSize[0], aParentDifferentialSize[1], aParentDifferentialSize[2])
            .setOrigin(0.5, 0.5, 0.5)
    ;

    return node;
}
</script>
</body>
</html>
```

**Comment by aprender** [comment]
@alexanderGugel tested against last night's develop branch (a5d7fd552505c0f2d40c4a86aa0a61950736ff16) problem still exists.
